WordPress/gutenberg
# DataViews: Redesign of filters

## What?
Part of: https://github.com/WordPress/gutenberg/issues/55100

This PR redesigns the DataViews filters based on: https://github.com/WordPress/gutenberg/issues/55100#issuecomment-1877128587.

Based on the designs we need a new widget, as we don't have something similar in the `@wordpress/components` package. In order to give us time to explore a good solution that can satisfy our use cases, the new component is added directly in the DataViews package. It uses some lower level components from `@ariakit`.


This PR needs polishing(mostly design) and feedback. 





### Tasks
- [ ] Fix/add e2e tests before merging


## Screenshots

https://github.com/WordPress/gutenberg/assets/16275880/112d4dae-f5f7-43e8-bc0b-327b893e9ea7



## Testing instructions
1. In pages that use DataViews(pages, templates, template parts, patterns) ensure that everything works as expected
2. Check that the primary filters API works as expected
3. a11y testing
